### PR TITLE
Change IP to string field

### PR DESCRIPTION
**Pull request by macfarla**
To match GoQuorum permissioning, change IP field from bytes16 to string. Eventually this can support DNS entries.

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by mark-terry — APPROVED** [review]
LGTM.

File: ethereum/permissioning/src/main/java/org/hyperledger/besu/ethereum/permissioning/NodeSmartContractV2PermissioningController.java [hunk 1]
```diff
@@ -64,14 +64,14 @@ private boolean isPermitted(final EnodeURL enode) {
   private Bytes createPayload(final EnodeURL enodeUrl) {
     try {
       final String hexNodeIdString = enodeUrl.getNodeId().toUnprefixedHexString();
-      final byte[] ip = encodeIp(enodeUrl.getIp());
+      final String hexIpString = encodeIp(enodeUrl.getIp());
```
[the remark this answers is not in the document]
Comment (macfarla) [reply ethereum/permissioning/src/main/java/org/hyperledger/besu/ethereum/permissioning/NodeSmartContractV2PermissioningController.java:67]: yep changed to just the string of the host name. Now we don't need the encodeIp method at all

[the remark this answers is not in the document]
Comment (lucassaldanha) [reply ethereum/permissioning/src/main/java/org/hyperledger/besu/ethereum/permissioning/NodeSmartContractV2PermissioningController.java:67]: Yep, it is probably better if we remove the `encodeIp` method and rename `hexIpString` variable to something like `address`.

**Review by lucassaldanha — APPROVED** [review]
LGTM. Only a suggestion for removing an unnecessary method and renaming a variable.

File: ethereum/permissioning/src/main/java/org/hyperledger/besu/ethereum/permissioning/NodeSmartContractV2PermissioningController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (macfarla) [reply ethereum/permissioning/src/main/java/org/hyperledger/besu/ethereum/permissioning/NodeSmartContractV2PermissioningController.java]: done

**Pull request merged**